SocketRocket/SRWebSocket.m

```diff
@@ -1378,90 +1417,98 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode;
     }
 
     dispatch_async(_workQueue, ^{
-        switch (eventCode) {
-            case NSStreamEventOpenCompleted: {
-                SRFastLog(@"NSStreamEventOpenCompleted %@", aStream);
-                if (self.readyState >= SR_CLOSING) {
-                    return;
-                }
-                assert(_readBuffer);
-                
-                if (self.readyState == SR_CONNECTING && aStream == _inputStream) {
-                    [self didConnect];
-                }
-                [self _pumpWriting];
-                [self _pumpScanner];
-                break;
+        [weakSelf safeHandleEvent:eventCode stream:aStream];
+    });
+}
+
+- (void)safeHandleEvent:(NSStreamEvent)eventCode stream:(NSStream *)aStream
+{
+    __strong typeof(self) strongSelf = self;
```

Comment:
I think you're 100% correct. As this PR was submitted over a year ago, I am having a hard time recalling the context, but this looks to me to be cruft that was left over from the modification.